src/extensions/__tests__/is-compatible-extension.test.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import semver from "semver";
+import {
+  isCompatibleExtension,
+} from "../extension-discovery/is-compatible-extension/is-compatible-extension";
+import type { LensExtensionManifest } from "../lens-extension";
+
+describe("Extension/App versions compatibility check", () => {
+  it("is compatible with exact version matching", () => {
+    expect(isCompatibleExtension({
+      appSemVer: semver.coerce("5.5.0"), // current app version
+    })(getExtensionManifestMock({
+      lensEngine: "5.5.0", // requested app version by extension ("semver"-format)
+    }))).toBeTruthy();
+  });
+
+  it("is compatible with higher patch-versions of main app", () => {
+    expect(isCompatibleExtension({
+      appSemVer: semver.coerce("5.5.5"), // for patch-versions
+    })(getExtensionManifestMock({
+      lensEngine: "^5.5.0",
+    }))).toBeTruthy();
+  });
+
+  it("supports short versions format for engines.lens", () => {
+    const isCompatible = isCompatibleExtension({
+      appSemVer: semver.coerce("5.5.0"),
+    })(getExtensionManifestMock({
+      lensEngine: "5.5",
+    }));
+
+    expect(isCompatible).toBeTruthy();
+  });
+
+  it("supporting `manifest.engines.lens='*'` to match any base-app version", () => {
+    expect(isCompatibleExtension({
+      appSemVer: semver.coerce("1.0.0"),
+    })(getExtensionManifestMock({
+      lensEngine: "*",
```

Comment:
I don't think we want to support `*`.

Reply:
Why not? If extension does really some simple things and doesn't really use lot of `@k8slens/extensions`-apis so then using star-value could be preferred. And why to restrict that anyway? It's responsibility for authors of extensions to keep them in sync with latest supported manifest versions for _lens-extension_.

Reply:
The main reason is that once auto update is turned on we might auto update over to a major version and then this will still say "compatible"